test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatPushTests.cs

```diff
@@ -188,6 +189,90 @@ public async Task PushMultiplePathsToFileSystemSource()
             }
         }
 
+        [Fact]
+        public async Task PushCommand_ConfigFile_ExpectedFormat()
+        {
+            var log = new TestCommandOutputLogger();
+            var configFileName = "My.Config";
+            var repositoryKey = "MySource";
+
+            using (var testDirectory = TestDirectory.Create())
+            {
+                FileInfo testPackageInfo = await TestPackagesCore.GetRuntimePackageAsync(testDirectory, "testPackageA", "1.1.0");
+                var repositoryPath = Path.Combine(testDirectory, "repository");
+                var configFilePath = Path.Combine(testDirectory, "config");
+                Directory.CreateDirectory(repositoryPath);
+                Directory.CreateDirectory(configFilePath);
+
+                FileSystemUtils.CreateFile(
+                    configFilePath,
+                    configFileName,
+                    string.Format(
+                        CultureInfo.InvariantCulture,
+                        @"<?xml version=""1.0"" encoding=""utf-8""?>
+                        <configuration>
+                            <packageSources>
+                                <add key=""{0}"" value=""{1}"" />
+                            </packageSources>
+                        </configuration>",
+                        repositoryKey, repositoryPath));
+
+                var pushArgs = new List<string>
+                {
+                    Command_Push,
+                    testPackageInfo.FullName,
+                    Param_Source,
+                    repositoryKey,
+                    Param_ConfigFile,
+                    Path.Combine(configFilePath, configFileName)
+                };
+
+                CommandLine.XPlat.Program.MainInternal(pushArgs.ToArray(), log);
+
+                ValidateSuccessfulRun(log);
+
+                Assert.True(
+                    File.Exists(Path.Combine(repositoryPath, testPackageInfo.FullName)),
+                    string.Format(
+                        CultureInfo.InvariantCulture,
+                        @"The package {0} was not found in {1}",
+                        testPackageInfo.FullName, repositoryPath));
+            }
+        }
+
+        [Fact]
+        public void PushCommand_ConfigFile_MissingName()
+        {
+            var log = new TestCommandOutputLogger();
+
+            var pushArgs = new string[]
+            {
+                Command_Push,
+                "testPackage1",
+                Param_ConfigFile
+            };
+
+            var exitCode = CommandLine.XPlat.Program.MainInternal(pushArgs, log);
+
+            Assert.True(
+                exitCode != 0,
+                "The run did not fail as desired. Simply got this output:" + log.ShowMessages());
```

Comment:
Another area where I'm not sure adding strings is helpful.

```suggestion
                log.ShowMessages());
```